.github/workflows/ci.yml

```diff
@@ -0,0 +1,46 @@
+name: CI
+
+on:
+  push:
+    branches:
+    - main
+  pull_request:
+
+env:
+  # The NAME makes it easier to copy/paste snippets from other CI configs
+  NAME: mir-json
+
+jobs:
+  lint:
+    runs-on: ubuntu-latest
+    steps:
+    - uses: actions/checkout@v4
+    - name: Deps
+      run: |
+        rustup default nightly-2023-01-23
+        rustup component add clippy rustc-dev
+    - uses: Swatinem/rust-cache@v2.7.0
```

Comment:
I wonder if we should configure the cache a bit more to make it behave like our typical Haskell GHA setups. After reading the [`rust-cache` README](https://github.com/Swatinem/rust-cache/blob/b1db5f9d5f9ed685de3a2774c3d3b838d2dde462/README.md) a bit, I think we should tweak the following:

* I think we should enable `cache-on-failure` so that build dependencies will still cache even if building `mir-json` itself fails.
* To future-proof this a little bit in case we accidentally corrupt the cache and need to invalidate it, I think it would be worth setting `prefix-key: ${{ env.CACHE_VERSION }}`, where [`CACHE_VERSION`](https://github.com/GaloisInc/cryptol/blob/d602858ea498360126633209d98367654d461995/.github/workflows/ci.yml#L19-L25) is a GHA environment variable.

Similarly for the use of `rust-cache` in the other job below.